### DISCUSSION TITLE
Welcome to Pinguino Discussions!

### DISCUSSION THREAD

**Discussion by MefhigosetH** [PinguinoIDE/pinguinoide.github.io#5] [Project Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

- - - - - - - - - - - - - - - - - - - - - - - - - - -

## Bienvenidos!
Comenzamos a utilizar Github Discussions como un lugar para conectar con otros miembros de nuestra comunidad. Esperamos que puedas:
* Publicar consultas acerca de los que te interesa.
* Comparir ideas.
* Conocer otros miembros de la comunidad.
* Dar la bienvenida a otros y ser de mente abierta. Recuerda que a esta comunidad la hacemos entre todos.

Para comenzar, comenta debajo de esta publicación, danos una introducción sobre vos y cuentanos qué haces con Pinguino.

**Comment by swhomans** [comment]
Hi Folks,

I'm a retired professor of biophysics. I have been using the Pinguino environment successfully for a number of years with my hobby which is amateur radio (you can see my projects at stevewhomans.wordpress.com). I recently tried installing Pinguino v13 using a fresh install of Ubuntu 20.04 running virtualbox 6.1 (windows host), python 3.8.5 and pips 20.0.2. I encountered a couple of trivial issues during the install that I thought I should share:

1. I needed to install libxcb-xinerama0 to get the GUI running
2. The .conf file generated by the install script does not point to the correct directory for the P32 libraries.

I resolved the second problem by editing the .conf file directly, but it might be useful to fix this error in the next version of the installer.

A further problem that I have not been able to fix concerns interrupts. I first noticed this with my code but it is also demonstrated by attempting to compile the example under "08.Interrupt" entitled "P32-FrequencyMeter80MHz.pde" (using Pinguino P32 OTG board). The linker complains that the ISR is too large. I can compile my own code (and the example just mentioned) with no problem using Pinguino X4 rev959 (on windows), but I have never been able to compile this code with later versions. Since X4 has been fine for my needs I have not pursued the problem, but it would be good to resolve it moving forwards perhaps. I'm not an expert but it seems to me that later versions are trying to insert and run the ISR code within the interrupt vector table space, rather than creating a pointer to the ISR. Does anyone have any thoughts? Many thanks! Regards, Steve H.

**Reply by MefhigosetH** [reply]
Dear @swhomans ,

Thanks for your support and feedback. Welcome to the Pinguino Community !

Yes, the P32 libraries on the Linux installer is a known bug and will be fixed on the next version.

About the interrupt issue, I understand what's the problem, but I'm not sure how to fix this now :-P Maybe someone on the community have a clue...

We'll be doing our best to upgrade the website, documentation and code. Let's do Pinguino the best OpenHardware platform, together.

Thanks again, and best regards.

**Reply by swhomans** [reply]
Many thanks. I’m trying to understand the interrupt issue myself. If anyone in the community has any ideas about how to fix this, please let me know - I have plenty of time to work through it. 

Do you have any idea what happened to the detailed documentation of functions, libraries etc in the previous pinguino.cc? If it still exists somewhere I’d be happy to work though it and update it.

**Reply by MefhigosetH** [reply]
It's seem that exists a copy of the old Pinguino Wiki on Web Archive Org:

https://web.archive.org/web/20140328010344/http://wiki.pinguino.cc/index.php/Language_Reference

It would be great if you could migrate that content to the new website !

**Reply by MefhigosetH** [reply]
If you don't know how to add content to the new site, I can teach you.

**Reply by swhomans** [reply]
Ok sure, I’ll start doing that next week. It will take some time I guess, but I have plenty of that being retired :-). I can see the archived content, so it will be straightforward to update it. I thought the original documentation was good, but for example it was sometimes unclear whether an entry referred to P8 or P32 (or both).  I’ll go through the text first, then figure out how to port it to the new site. I might indeed need you help with this last stage. I think it would worthwhile working through the example sketches too, checking whether they compile and so on...

**Reply by funlw65** [reply]
What happened with the old site? Who is behind the new site, reasons, plans for the future? It would be nice to present a history about the original author (on pinguino.cc), what happened and what to expect from now on.

**Reply by swhomans** [reply]
Hi @funlw65, I'm not certain what happened to the old site or  the original authors. You can find some earlier  information at jpmandon.blogspot.com.  I have been a user of Pinguino for over a decade, and I know that J P Mandon was originally involved in the project. You can also find much of the original website as described in an earlier post above. I've been updating this information for the new site. I am not one of the developers though, and hopefully they will be able to update you on future expectations. Meantime I have a reasonable degree of experience with Pinguino (particularly P32), so I'd be happy to help if you have any issues.

**Comment by IdearDabry** [comment]
### ¡Ayuda Urgente! Para poder Usar **Pinguino IDE-v13** en **Linux Debian-Buster 10**
¡Hola a todos, tengan un buen día! Mi nombre es DaBry. Necesito su ayuda para poder usar el **Pinguino IDE-V13**
No se, si este Hilo sea el más apropiado, o si tal vez deba abrir un hilo con mi tema aparte; pero 
les escribo este comentario; para solicitar su Ayuda, con la Instalación y Uso del  **Pinguino IDE-v13**,  es decir; Yo soy un Usuario Linux desde hace unos pocos años, y ya no uso el OS de Windows Microsoft, ni tampoco quiero volver a usarlo nunca más! Jajajajaja, por lo menos No, mientras exista GNU-Linux. El asunto es que; descargue de su Sitio Web Oficial el (**installer-v2.sh**)  para **Pinguino IDE-v13**  lo instale según todas las instrucciones, en mi computador, el cual tiene un **Sistema Operativo GNU-Linux_Distribución Q4OS con Debian Buster-10**  y un entorno de escritorio  **KDE Plasma**  Al abrir la Consola, para intentar instalar el **Pinguino IDE-v13** por medio del archivo  **Installer-v2.sh**  aparecen unas dependencias que faltaban por instalar;  pedía instalar las siguientes dependencias: python3, pip, pipenv. Entonces inmediatamente hice lo indicado, instale por medio de la Consola las dependencias solicitadas, quedaron perfectamente bien instaladas y aceptadas por mi sistema operativo; y luego por Consola, regrese a la instalación del  **Pinguino IDE-v13**  con el archivo:   **installer-v2.sh**   según la información arrojada en la pantalla de Consola, todo quedo muy bien instalado, y sin conflictos.

![Instalacion(installer-v2 sh)-Pinguino IDE-v13_Foto1](https://user-images.githubusercontent.com/96216143/146301175-fee9ca55-ac9b-4623-900f-2cfd0172a9d7.png)
![Instalacion(installer-v2 sh)-Pinguino IDE-v13_Foto2](https://user-images.githubusercontent.com/96216143/146301202-e645f7d8-8fae-4601-9835-222fae2fff09.png)

Luego de finalizar la instalación, busco el icono de la aplicación **Pinguino IDE**,  le doy todos los Permisos Necesarios de Usuario; para que se Ejecute como programa, pero cuando le doy doble click al icono de **Pinguino IDE** para  abrir, parece intentar abrir; pero No, el programa No hace nada, No abre, No se ejecuta, No abre nada, No muestra nada, es decir; 

![2021-12-13_05-53](https://user-images.githubusercontent.com/96216143/146301477-88be1b7d-ac26-4cf1-bff7-b4456f76ee5e.png)

**No abre ninguna ventana GUI de interfaz de Usuario**. Como no abre la GUI interfaz de Usuario de  **Pinguino IDE versión-13**   verifique tener instalada la dependencia **libxcb-xinerama0**,  por si acaso; fuese ese el problema, pero ya está instalada; incluso también instale la dependencia **libxcb-xinerama0-Dev**  pero tampoco funciono. Al no abrir con el Icono de **Pinguino IDE**  en el Escritorio, luego intente abrir la Aplicación del Programa **Pinguino IDE**, Directamente por medio de Consola...

![Intento de Inicio Pinguino IDE-v13_Foto3](https://user-images.githubusercontent.com/96216143/146302080-5db1514d-0b7f-4800-a753-ca2b0085a27e.png) pero tampoco abre. Tambien lo intente por medio de la Consola, tratando de forzarle; usando la opción de Sudo...

![Intento de Inicio Pinguino IDE-v13 con Sudo_Foto4](https://user-images.githubusercontent.com/96216143/146302506-29e83a87-5927-4a21-adfe-b765983801ca.png)

pero tampoco abre, ** por lo menos No en modo Gráfico de una Interfaz de Usuario Gui**; aunque al parecer, el programa si está operable de alguna manera, porque según alcanzo a entender, la pantalla de consola arroja información; acerca de unos módulos y un módulo USB.
La verdad ya me tiene desesperado y no se que mas hacer para que funcione! Por eso acudo a Ustedes y toda la comunidad del **Proyecto Pinguino IDE**; para solicitar su ayuda con este problema y poder encontrar una oportuna solución!
Junto con esta Solicitud de Ayuda Escrita, les envío también Imágenes captura de Pantalla, de toda la cuestión. Espero me puedan Ayudar lo mas Pronto y Rápido Posible; porque necesito usar el  **Proyecto Pinguino IDE**,  para programar un  **Micro-Controlador 18f2550**  y poder Desarrollar un Proyecto de Electrónica, para darlo a conocer; a una comunidad de estudiantes que están muy entusiasmados con la **Idea del Hadware y Software Libre; en Colombia Sur América**.  Antes cuando Yo Usaba OS de Windows Microsoft, utilizaba los microcontroladores de PICAXE pero desde que apareció Arduino, y su Homólogo  **Proyecto Pinguino IDE**  100% libre, me parece Mucho Mejor la Idea; de Crear, Desarrollar y Usar Proyectos de Electrónica de Control con Micro-Controladores, 100% libres; como lo es la filosofía del  **Proyecto Pinguino IDE**. 
Desde ya, muchas gracias por su atención. ¡Quedo Atento y Pendiente!

**Reply by IdearDabry** [reply]
Hola a todos, les comento, que Afortunadamente, he recibido Rápida y Oportuna Ayuda con este problema arriba mencionado. Ayuda y Atención de Miembros de la Comunidad Linux, de aquí mismo de la Comunidad del Proyecto Pinguino IDE. Contacte por correo electrónico,a el Amigo Víctor Villareal, de Argentina; directamente involucrado con el actual Proyecto Pinguino IDE, y el amigo Cristian Paul Peñaranda, de Colombia.
Entonces para hacer un buen y claro contexto, para que esta información sirva a cualquier otra persona usuario; que le pueda estar ocurriendo lo mismo o algo similar, muy en especial para usuarios que recién como Yo; vienen migrando de Windows microsoft, a GNU/Linux. A todos esos Usuarios que vienen de Windows, que están "Recién llegado y Nuevos" en GNU/Linux les conviene mucho leer todo esto!... Comentare a continuación y en detalle; todo lo que se hizo, para buscar la solución a este Problema!....
Al parecer, en algunas ocasiones, se presenta una falla en el funcionar de alguno y otro programa, con la aparente falla en la ejecución de su GUI o interfaz de Usuario; en Sistemas Operativos GNU/Linux, con Debian, por ejemplo en Debian Buster-10 ocurre, o puede que llegue a ocurrir, que alguna GUI del archivo y/o icono ejecutable de algún programa y aplicación, que no son del repositorio oficial de las distribuciones, sino que pertenecen a los desarrollos programas y aplicaciones free independientes, algunas veces estas no abren (No se ejecutan); o por lo menos no gráficamente en un entorno de ventana, ni siquiera por medio de la consola, incluso ni si usamos sudo, en su ruta de directorio de ubicación hacia el ejecutable. Indicativo este; de que algo no esta bien con alguna librería, dependencia no satisfecha, o algún archivo o carpeta de configuración y uso, que no esta visible, casi siempre en los ficheros que no son directamente del usuario, pero que si son de la Raíz del Sistema Operativo. Lo cual No significa que nuestro PC vaya a explotar por eso, jajajajajajaja, tampoco significa que nuestro Sistema operativo este dañado, ni es nada realmente grave, pero si en cualquiera de sus modos; es un verdadero fastidio, un problema menor, pero un dolor de cabeza en muchos casos.
Eso a mi me a sucedido en varias ocasiones, con uno que otro programa y/o aplicación, en especial me paso con la aplicación del Proyecto Pinguino IDE, y su Ejecutable de Entorno Gráfico GUI; como no se abría, al ejecutar el Icono del Programa, ni tampoco abría, al ejecutar la ruta del Archivo pinguino.sh entonces tuve que recurrir a la comunidad Linux en internet, en especial; a los que mas saben del desarrollo, mantenimiento y operación del Proyecto Pinguino IDE. (Entronorno de Open Software y Open Hardware, con Tarjetas de Desarrollo Electrónico similar a Arduino; basado en Microcontroladores de 8 y 32 bits de la compañía Microchip, con sistema USB integrado).
Así las cosas, me Ayudaron muy Rápidamente con el Problema, el Amigo Victor Villareal, de Argentina; directamente involucrado con el actual Proyecto Pinguino IDE, y el amigo Cristian Paul Peñaranda, de Colombia. Puntos de Estrellas para ellos!
Entonces según la guía y el consejo de estos estimados amigos y conocedores de GNU/Linux y del Proyecto Pinguino IDE; en fallas similares a esta, lo primero es verificar cual es realmente el problema! Lo cual podemos intentar averiguar abriendo nuestra Terminal de Consola de Usuario Linux, y escribir lo siguiente sin sudo; así tal cual:

export QT_DEBUG_PLUGINS=1

Luego según las recomendaciones del sitio web: [https://askubuntu.com/a/1054691](url) que recomienda seguido a eso; según las condiciones y características del problema en si, llamar o invocar por la misma Terminal y Consola de usurario, en la que hemos hecho lo anterior, osea escribimos el nombre del archivo ejecutable; de la "supuesta aplicación rebelde" que, aparentemente esta fallando al no abrir. En mi caso y problema especifico de este ejemplo, el archivo ejeuctable (.sh) del programa pinguino IDE, con todo y su ruta completa de ubicación, es decir:

/home/Usuario/.pinguino/v13/pinguino-ide-fix-linux-support/pinguino.sh

(La ruta puede cambiar, según el nombre Usuario de cada quien; y de como tenga configurada su instalación de su Sistema Operativo). Pero en general vendría siendo algo muy similar a eso!...Una vez hayamos tecleado debidamente, todo lo anterior correspondiente, según a este asunto, entonces el sistema Operativo hará un análisis de donde, o en que podría consistir la falla!....Esperamos y después de un poco tiempo, nos arrojara una información de salida, por la misma pantalla de la Terminal Consola de Usuario; Información esta que tal vez puede llegar ser muy extensa, por lo que habría que detenerse con calma a leer y analizarla, para esto ultimo; si nosotros no somos muy conocedores del tema, conviene mucho pedir la ayuda de personas Expertas en el tema; como en este caso lo son los amigos y colaboradores arriba mencionados. o también acudir a todos los foros de discusión temática en Internet, como el sitio, web arriba mencionado, o el que mencionaré mas adelante.
Bueno lo importante y para resumir, es que en mi caso comprobado y especifico de la falla de no apertura y ejecución del Pinguino IDE, pues según el consejo del Amigo Victor Villareal de Argentina, del Proyecto Pinguino IDE, quien al analizar la información del problema, arrojada por Consola; Acertó puntualmente al decir o Comentar Textualmente que:  

> "pyQT quiere acceder al módulo "libxcb-util.so.1" pero no lo encuentra. Googleando un poco, resulta que Debian Buster solo viene con el módulo "libxcb-util.so.0". Y no la versión 1. Cómo workaround, se recomienda crear un symlink o enlace simbólico "libxcb-util.so.1" que apunte a tu actual "libxcb-util.so.0"  Así lo podemos ver claramente en el siguiente Foro de este Sitio Web: 

[https://github.com/tasmota/tasmotizer/issues/71](url) 
allí en ese foro de análisis y discusión temático de Problemas; podemos leer Textualmente lo Siguiente:

> "Mrnt (Comento):
"Parece que a partir de PyQt5 5.15, el complemento Qt libqxcb.soespera encontrar el módulo libxcb-util.so.1p.ej:
/lib/x86_64-linux-gnu/libxcb-util.so.1
Sin embargo, Debian Buster (versión 10.7) solo tiene libxcb-util.so.0
Este cambio aparentemente se realizó en PyQt 5.15.
Solución posible:
Una posible solución podría ser fijar el requisito máximo para PyQt5 en 5.14.2, aunque no lo he probado (opté por hacer un enlace simbólico llamado libxcb-util.so.1en la misma carpeta que libxcb-util.so.0eso apuntaba a libxcb-util.so.0, que parece funcionar), por ejemplo:

sudo ln -s /usr/lib/x86_64-linux-gnu/libxcb-util.so.0 /usr/lib/x86_64-linux-gnu/libxcb-util.so.1 "

Después de leer todo esto, y lo comentado en ese foro, en donde algunos usuarios, que han tenido fallas similares; en el Inicio y/o Apertura, con otros programas y aplicaciones quienes según los comentarios, siguieron esa recomendación, el problema se resolvió y desapareció la falla. Así las cosas parece estar clara la solución!
Entonces según todo lo anterior, Abrimos nuestra Terminal Consola de Usuario, y tecleamos o copiamos exactamente lo siguiente:

sudo ln -s /usr/lib/x86_64-linux-gnu/libxcb-util.so.0 /usr/lib/x86_64-linux-gnu/libxcb-util.so.1

Eso creará algo así como un symlink o enlace simbólico llamado "libxcb-util.so.1" en el mismo sitio o carpeta; que apunte o apuntaba a la actual "libxcb-util.so.0"
Yo en mi caso, hice exactamente eso, por medio de la Terminal Consola de Usuario; y ¡EUREKA...ALBRICIAS!...Al abrir el icono y/o archivo ejecutable, de pinguino IDE, ¡se abre y se ejecuta de inmediato!...Problema Solucionado! Ya puedo Usar el programa y todas las grandes Ventajas del Proyecto Pinguino IDE.
![Por fin Funciono-Abrio-pinguino IDE_Gracias](https://user-images.githubusercontent.com/96216143/146527322-cfc6b4f9-76cc-44f3-8052-d2b537289fd6.png)
![Pinguino IDE](https://user-images.githubusercontent.com/96216143/146527407-bcf69391-7b18-40ac-894a-1ddaf6d8591f.png)

Con lo cual doy por Resuelto, y Finalizado este Tema.
¡Muchas Gracias a Todas las Personas de la Comunidad; por su Ayuda en la Resolución Rápida y Eficaz de este Inconveniente!...Felices Fiestas, Feliz Navidad, Saludos y Bendiciones para todos!
